addon/services/session.js

```diff
@@ -200,6 +200,13 @@ export default Service.extend(Evented, {
     {{#crossLink "SessionService/invalidationSucceeded:event"}}{{/crossLink}}
     event.
 
+    Please note that a session which is already unauthenticated should not be
+    invalidated with this method. Instead make sure to call the invalidate
+    method only on a session which you have verified to be authenticated,
+    f.e. by checking if the session.isAuthenticated property is true. See also
+    {{#crossLink "SessionService/isAuthenticated:property"}}{{/crossLink}})
```

Comment:
Is this actually true though? I think the reason for removing the assertion was that we figured it doesn't really matter whether the session is authenticated when calling `invalidate` as it will always be invalidated in the end anyway.

Reply:
According to the discussion, calling invalidate on an already unauthenticated session raises an error or assertion depending on the environment and developers are surprised that they get an error in production in the scenario of trying to invalidate an already unauthenticated session whereas in `dev` mode the assertion prevents this error to be thrown. 

One of the final suggestions was to remove the assertion (so the error becomes visible in `dev` mode already) and to update the docs accordingly:

 
https://github.com/simplabs/ember-simple-auth/issues/1425#issuecomment-324565456: 
>> I think it would make sense to explicitly state in docs that invalidate method should not be called on unauthenticated session
> Yeah, that sounds like a good idea. PR would be welcome!

https://github.com/simplabs/ember-simple-auth/issues/1425#issuecomment-344175432
>> It would make it easier for developers if invalidate just returned immediately if !isAuthenticated

> I strongly believe this would make the API much harder to understand and use - returning a promise in one scenario and returning immediately in another scenario means you have to check which scenario you're in before you can even call the method as you have to handle the returned value totally different in the 2 scenarios.

https://github.com/simplabs/ember-simple-auth/issues/1425#issuecomment-344183193
> Yeah, I agree removing the assertion might be a good idea. I guess removing an assertion should also not be considered a breaking change?
> This is a good first issue as the only thing that needs to be done is to remove the assertion (and potentially a test that asserts the assertion exists :) ).

From the [original issue discussion](https://github.com/simplabs/ember-simple-auth/issues/1425)

Do you think another approach might make more sense @marcoow ? One of the suggestions in the discussion was to already return a resolved Promise from the `invalidate` hook in case the session is already unauthenticated, making it possible for the end user to not having to worry about the authentication state before calling the method. 
I'd assume this would be a breaking change of the API though as you already mentioned in the discussion yourself but I also think it makes the API more convenient.

Reply:
I think the simplest fix for this is probably if we just add

```js
if (this.get('session.isAuthenticated')) {
  return RSVP.resolve();
}
```

to this method. That way we can remove the assertion, will never run into the original error reported in #1425 and can also keep the guarantee for authenticators that the session will always be authenticated and session data present when the __authenticator's__ `invalidate` method is called.

What do you think?

Reply:
Yes, that sounds good - I think it would check the session to be unauthenticated though, correct?

```
if (!this.get('session.isAuthenticated')) {
  return RSVP.resolve();
}
```

I can update the PR accordingly then

Reply:
Yes, of course it should check for unauthenticated 🤦‍♂️